tools/castle-editor/code/framedesign.pas

```diff
@@ -2026,6 +2028,24 @@ procedure TDesignFrame.PropertyGridModified(Sender: TObject);
   MarkModified;
 end;
 
+procedure TDesignFrame.PropertyEditorModified(Sender: TObject);
+var
+  Sel: TComponent;
+begin
+  Sel := SelectedComponent;
+  if Sender is TPropertyEditor then
+  begin
+    if Sel <> nil then
```

Comment:
Note that `Sel` is _almost always_ `nil` here in all my tests and `TPropertyEditor(Sender).GetName` contains the name of the component that was changed. `Sel <> nil` happens only in case of the value come from `TDialog` AFAIR and in that case - and `TPropertyEditor(Sender).GetName` doesn't contain the name of the component in this case, so that they supplement each other.

I'll try `PropCount`.

Reply:
> in all my tests and TPropertyEditor(Sender).GetName contains the name of the component that was changed

Note that `TPropertyEditor(Sender).GetName` should definitely be a name of the property, not the component. We even override it sometimes (see `TCastleVector2PropertyEditor.GetName`) assuming that.

Reply:
> Note that TPropertyEditor(Sender).GetName should definitely be a name of the property, not the component.

I have no idea how I got that idea... I clearly remember I had it like "Change ColorRectangle1.Color to FFFFFF" - but now I can't repeat that in any way :D Maybe I was too tired and sleepy and my eyes got blurry? Nah, I clearly remember that because it really surprised me. Anyhow, now it's working exactly as you've written, sorry for the noise :)

Fixed in 9af1ba229c0d4439723ae1e30666362fa0bdd076

Reply:
Ah, now I get it... sometimes both `PropertyGridModified` and `PropertyEditorModified` are called and when (before) it had "ucLow" it would overwrite the undo message with `PropertyGridModified`'s one which indeed is constructed as `SelectedName.Property`.